pkg/securelaunch/policy/vm_test.go

```diff
@@ -0,0 +1,223 @@
+	"7e5398dcf2b2510dcde1fb3bb1a7e0c17e00683a0dbfd53ad4170b2ac24741b7" +
+	"dfff09ab7b6a960e7f6937ff85b6289f9a1e83a1abeb653ae0aa2682809b6242" +
+	"2adb0130cad9ba9db54e0af7be0b5b008dcb4850e049a7b3b238f8feb5bbc91e" +
+	"97c55e3becaf7a1fec21aca19a1bb6d283c8c7848686746344ed1481a57154d2" +
+	"56c20efdfe0121097734222d02f0529372ab7ab296ec0b244e830c5540270e7f" +
+	"4fa729ec35328f63f83b3e6494187c254372261970764fb99438976d904bcffb" +
+	"1bb9e0428cc03e50595f07273fc387a4df1611275bc82fb156d9bf0ce8bbd0b7" +
+	"f6e008835c9f701e25afd6a531e7c8ab354c3ff9672a5354306091f919fa46bf" +
+	"8d3ca668f837ccf7a3adc6841c0c416ecafd299de3c5e03cff28daad9efd5c2a" +
+	"2de4526869a209fa6e5703ea6b63d40cb534e1f192458978c475dd155d919fe8" +
+	"ff16685d488051f3618b4a3112154a12206e321299b6e0a6a4ddff623a2f4f08" +
+	"4d254392e9285aa245242505bb2d94e6d6fd4ada8d12a6ffd3440227e17771e6" +
+	"68404e53b87a8a74314f70a8bc710c83459932357d068ad62cd5feeacae6f5c1"
+
+// VM setup:
+//
+//	 /dev/sda is ../testdata/mbrdisk
+//	 it must be copied because there is another test that uses it and
+//	 qemu will fail if they run at the same time.
+//		  /dev/sda1 is ext4
+//		  /dev/sda2 is vfat
+//		  /dev/sda3 is fat32
+//		  /dev/sda4 is xfs
+//
+//	  ARM tests will load drives as virtio-blk devices (/dev/vd*)
+const mbrFile = "../testdata/mbrdisk"
+
+func mbr(t *testing.T) (string, error) {
+	t.Helper()
+	t.Skipf("fix me")
+
+	d := t.TempDir()
+
+	b, err := os.ReadFile(mbrFile)
+	if err != nil {
+		return "", fmt.Errorf("reading  MBR file for policy:%w", err)
+	}
+
+	disk := filepath.Join(d, "disk")
+	if err := os.WriteFile(disk, b, 0666); err != nil {
+		return "", fmt.Errorf("writing MBR file for policy:%w", err)
+	}
+
+	return disk, nil
+}
+
+func TestVM(t *testing.T) {
+	qemu.SkipIfNotArch(t, qemu.ArchAMD64)
+	t.Skipf("fix me")
+
+	disk, err := mbr(t)
+	if err != nil {
+		t.Fatalf("create mbr disk file: got %v, want nil", err)
+	}
+
+	govmtest.Run(t, "vmpolicy",
+		govmtest.WithPackageToTest("github.com/u-root/u-root/pkg/securelaunch/policy"),
+		govmtest.WithQEMUFn(
+			qemu.WithVMTimeout(2*time.Minute),
+			// CONFIG_ATA_PIIX is required for this option to work.
+			qemu.ArbitraryArgs("-hda", disk),
+
+			// With NVMe devices enabled, kernel crashes when not using q35 machine model.
+			qemu.ArbitraryArgs("-machine", "q35"),
+		),
+	)
+}
+
+func TestParse(t *testing.T) {
+	guest.SkipIfNotInVM(t)
+	t.Skipf("fix me")
+
+	policyFile := "sda1:" + "/policy"
+
+	if err := slaunch.WriteFile([]byte(policyStr), policyFile); err != nil {
+		t.Fatalf(`WriteFile(policyStr.bytes, tempFile) = %v, not nil`, err)
+	}
+
+	if err := Load(policyFile, "", ""); err != nil {
+		t.Fatalf("Load() = %v, not nil", err)
+	}
+
+	policy, err := Parse()
+	if err != nil {
+		t.Fatalf("Parse() = %v, not nil", err)
+	}
+	if policy == nil {
+		t.Fatalf("no policy file returned")
+	}
+}
+
+func TestVerify(t *testing.T) {
+	guest.SkipIfNotInVM(t)
+	t.Skipf("fix me")
+
+	policyFile := "sda1:" + "/securelaunch.policy"
```

Comment:
Why string concatenation is necessary here? The string is already short.

Reply:
I think the author did this for clarity, but I really don't know. The author is unreachable atm.